src/layout/monitor.rs

```diff
@@ -598,16 +598,20 @@ impl<W: LayoutElement> Monitor<W> {
         }
     }
 
-    pub fn focus_window_or_workspace_down(&mut self) {
+    pub fn focus_window_or_workspace_down(&mut self) -> bool {
         if !self.active_workspace().focus_down() {
             self.switch_workspace_down();
+            return true;
```

Comment:
As far as I can tell, this will always return true regardless of whether the workspace is the last one, so it will never focus a different output. Did you test these actions?

Reply:
I have tested it and I just had bugs with the behavior, maybe this is it, I haven't had time to test it again yet 😅